File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        value_to_index = {}
        for i in range(0, len(nums)):
            if nums[i] not in value_to_index:
                value_to_index[nums[i]] = i
            
        print(value_to_index)
                
        for i in range(0, len(nums)):
            complacent = target - nums[i]
            if complacent in value_to_index and i != value_to_index[complacent]:
                return [i, value_to_index[complacent]]

File: 0001-two-sum/0001-two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        """
        {

        }
        """
        var complements: [Int: Int] = [:]
        for i in 0..<nums.count {
            if complements[nums[i]] == nil {
                complements[nums[i]] = i
            }
        }

        for j in 0..<nums.count {
            let complement = target - nums[j]
            guard let i = complements[complement] else { 
                continue
            }
            if i != j {
                return [i, j]
            }
        }
        return [] //will never reach
    }
}